azure-iot-device/tests/iothub/aio/test_async_inbox.py

```diff
@@ -26,17 +29,52 @@ def test_check_item_is_in_inbox(self, mocker):
         inbox._put(item)
         assert item in inbox
 
-    # this test also raises runtime warning if it's asynchronous
-    def test__put_adds_item_to_inbox(self, mocker):
+    @pytest.mark.it("Can be checked for whether or not it is empty")
+    @pytest.mark.asyncio
```

Comment:
I'm thinking out loud.  Do not change this now, but this phrase is a smell to me that says that the test class needs to be refactored.  "Can be checked" is not an action verb.

Reply:
Well, actually, it is a verb - "be" - the most forgotten verb. But I do agree the phrasing is SUPER awkward.

My frustration here (and in similar tests) is that this doesn't really feel like it should be a separate unit - it's validating a feature of the object itself, that relates to the properties of the object, and thus doesn't feel like an separate operation/unit. The fact that facts about the unit can be exposed is a requirement of the object IMO.

I think I'm missing a good word or two that might help. I'll revisit this in the morning with a refreshed mind.


---
In reply to: [291185176](https://github.com/Azure/azure-iot-sdk-python-preview/pull/101#discussion_r291185176) [](ancestors = 291185176)